fdb-record-layer-lucene/src/main/resources/META-INF/services/org.apache.lucene.codecs.Codec

```diff
@@ -0,0 +1,16 @@
+#  Licensed to the Apache Software Foundation (ASF) under one or more
+#  contributor license agreements.  See the NOTICE file distributed with
+#  this work for additional information regarding copyright ownership.
+#  The ASF licenses this file to You under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with
+#  the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+com.apple.foundationdb.record.lucene.codec.LuceneOptimizedCodec
```

Comment:
We also use `@AutoService` in this project to create these service files automatically, like this one for `LuceneKeyExpressionVisitorFactory`: https://github.com/FoundationDB/fdb-record-layer/blob/a16496c56a5afbb6806ea3011409f211474ea3ae/fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneKeyExpressionVisitorFactory.java#L26

Which is probably preferable if we can use it.

Reply:
Removed this file and added the annotation ```@AutoService(Codec.class)``` for ```LuceneOptimizedCodec```. @alecgrieser please let me know if this looks good to you.